include/wil/filesystem.h

```diff
@@ -83,7 +85,7 @@ namespace wil
         return false;
     }
 
-#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && (_WIN32_WINNT >= _WIN32_WINNT_WIN7)

```

Comment:
This seems like it excludes much more than it needs to

Reply:
I don't disagree - initially I wanted to only exclude functions which really relied on unsupported features, but I had an impression this would throw this header file into a spiral of ifdefs, as a lot of things in it seems to depend on each other, for example:

- `try_get_parent_path_range` and `GetFullPathNameW` rely on `find_last_path_segment`
- `CreateDirectoryDeepNoThrow` relies on `try_get_parent_path_range`
- `RemoveDirectoryRecursiveNoThrow` relies on `GetFullPathNameW`

I can of course spend some time ifdefing out functions in a more fine-grained fashion - but I am not convinced it makes sense. Both are fine by me.

Reply:
In general, we need to be careful about being too exclusionary for things like this. For example, because MSVC is historically so permissive, there's the chance that someone could be relying on a function that doesn't make use of any offending symbol. That said, I think we're likely fine here. (1) The union of projects building <= Vista and projects using WIL is likely very low, and (2) I _believe_ that the permissive issues generally apply to templates. Since there are offending functions that aren't templated, I have a strong feeling that anyone building <= Vista would receive build errors even under permissive mode.